### PR TITLE
Make the "Description" column in Code Styles Options a "dataitem" control

### PR DESCRIPTION
**Pull request by dpoeschl**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems?id=391567

### CODE REVIEW

**Comment by dpoeschl** [comment]
Tagging @dotnet/roslyn-ide for review.

File: src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml [hunk 1]
```diff
@@ -87,23 +87,25 @@
...
-                            <Setter Property="VerticalAlignment" Value="Center"/>
-                            <Setter Property="HorizontalAlignment" Value="Left" />
-                            <Setter Property="TextWrapping" Value="Wrap" />
-                            <Setter Property="AutomationProperties.Name" Value="{Binding GroupName}" />
-                        </Style>
-                    </DataGridTextColumn.ElementStyle>
-                </DataGridTextColumn>
+                    <DataGridTemplateColumn.CellTemplate>
+                        <DataTemplate>
+                            <local:TextBlockWithDataItemControlType
```
Comment (dpoeschl) [inline src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml]: Left everything the same as the "Setter" elements before, with the exceptions of "Text", "AutomationProperties.Name", and "Focusable".

File: src/VisualStudio/Core/Impl/Options/Style/AbstractCodeStyleOptionViewModel.cs [hunk 2]
```diff
@@ -56,6 +57,7 @@ public virtual NotificationOptionViewModel SelectedNotificationPreference
             Preferences = preferences ?? GetDefaultPreferences();
             NotificationPreferences = notificationPreferences ?? GetDefaultNotifications();
             GroupName = groupName;
+            GroupNameAndDescription = $"{groupName}, {description}";
```
Comment (dpoeschl) [inline src/VisualStudio/Core/Impl/Options/Style/AbstractCodeStyleOptionViewModel.cs]: We were getting around concatenating these before in a seriously ugly way. I can explain if needed. This is much better.

**Comment by sharwell** [comment]
Here's a copy of the original bug report for future readers that don't have access:

> **Steps to reproduce:**
> 1. Open Visual Studio.
> 2. Go to Tools-> Options -> Text Editor-> c#-> Code Style-> General.
> 3. Navigate to table RHS pane and observe  "LocalizedControlType"  property for data items under "Description" column header.
>
> **Actual Result:**
"LocalizedControlType"  property for data items under "Description" column header is "Custom".
**Narrator Behavior:**
Narrator is announcing as "Table, For locals,parameters and members ,custom, Column Header Description."
>
> **Expected Result:**
"LocalizedControlType"  property for data items under "Description" should be "data item".
>
> **User Impact:**
Visually impaired user who rely on screen reader will get confused with the inappropriate information.

**Pull request merged**